tests/components/deconz/test_fan.py

```diff
@@ -100,6 +262,16 @@ async def test_fans(hass, aioclient_mock, mock_deconz_websocket):
     )
     assert aioclient_mock.mock_calls[2][2] == {"speed": 0}
 
+    # Service turn on fan to low speed
+
+    await hass.services.async_call(
+        FAN_DOMAIN,
+        SERVICE_TURN_ON,
+        {ATTR_ENTITY_ID: "fan.ceiling_fan", ATTR_SPEED: SPEED_LOW},
+        blocking=True,
+    )
+    assert aioclient_mock.mock_calls[3][2] == {"speed": 2}
```

Comment:
speed 1 was expected here

Reply:
Yes, but I cant fix it without overriding the async_turn_on_compat since the calculations are run there

Reply:
Fixed and increased tests